src/sys/socket/addr.rs

```diff
@@ -882,6 +882,29 @@ impl SockAddr {
             SockAddr::Link(LinkAddr(ref ether_addr)) => (mem::transmute(ether_addr), mem::size_of::<libc::sockaddr_dl>() as libc::socklen_t),
         }
     }
+
+    pub unsafe fn as_ffi_pair_mut(&mut self) -> (&mut libc::sockaddr, libc::socklen_t) {
```

Comment:
This is duplicative.  Can you think of any way to share code between this method and `as_ffi_pair`?  Also, you should wrap to 80 cols.